node/forest_libp2p/src/service.rs

```diff
@@ -64,37 +67,26 @@ pub struct Libp2pService {
 
 impl Libp2pService {
     /// Constructs a Libp2pService
-    pub fn new(config: &Libp2pConfig, net_keypair: Keypair) -> Self {
+    pub fn new(config: Libp2pConfig, net_keypair: Keypair, network_name: &str) -> Self {
         let peer_id = PeerId::from(net_keypair.public());
 
         let transport = build_transport(net_keypair.clone());
 
         let mut swarm = {
-            let be = ForestBehaviour::new(&net_keypair);
+            let be = ForestBehaviour::new(&net_keypair, &config, network_name);
             Swarm::new(transport, be, peer_id)
         };
 
-        for node in &config.bootstrap_peers {
-            match node.parse() {
-                Ok(to_dial) => match Swarm::dial_addr(&mut swarm, to_dial) {
-                    Ok(_) => debug!("Dialed {:?}", node),
-                    Err(e) => warn!("Dial {:?} failed: {:?}", node, e),
-                },
-                Err(err) => warn!("Failed to parse address to dial: {:?}", err),
-            }
+        Swarm::listen_on(&mut swarm, config.listening_multiaddr).unwrap();
+
+        // Subscribe to gossipsub topics with the network name suffix
+        for topic in PUBSUB_TOPICS.iter() {
+            swarm.subscribe(Topic::new(format!("{}/{}", topic, network_name)));
```

Comment:
Not needed but could also be a one liner by changing to 

PUBSUB_TOPICS.iter().for_each(|topic|swarm.subscribe(Topic::new(format!("{}/{}", topic, network_name))));

Reply:
ends up being the same because the subscribe returns a boolean for if it was successful (ignoring because it should always be)